pkgs/by-name/di/dita-ot/package.nix

```diff
@@ -0,0 +1,55 @@
+{ stdenv
+, fetchzip
+, bash
+, jdk17
+, lib
+, makeWrapper
+, testers
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  name = "dita-ot";
+  version = "4.2.1";
+
+  nativeBuildInputs = [
+    makeWrapper
+  ];
+  buildInputs = [
+    bash
+    jdk17
+  ];
+
+  src = fetchzip {
+    url = "https://github.com/dita-ot/dita-ot/releases/download/${finalAttrs.version}/dita-ot-${finalAttrs.version}.zip";
+    hash = "sha256-PtscKsQwW1uBu0OfVHi7/d9JujGx12nZT+W3uMPQQWc=";
+  };
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/bin $out/$name/
+    cp -r $src/* $out/$name/
+
+    makeWrapper "$out/$name/bin/dita" "$out/bin/dita" \
+      --prefix PATH : "${lib.makeBinPath [ jdk17 ]}" \
+
+    runHook postInstall
+  '';
+
+  passthru.tests.version = testers.testVersion {
+    package = finalAttrs.finalPackage;
+    command = "dita --version";
+    version = "${finalAttrs.version}";
```

Comment:
You can omit these if you like, [defaults](https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/testers/default.nix#L60-L61) should work in this case.

Reply:
You are right :) I was not sure whether that was possible. Implemented 👍🏼